dallinger/recruiters.py

```diff
@@ -77,6 +81,11 @@ class PsiTurkRecruiter(Recruiter):
 
     def __init__(self):
         """Set up the connection to MTurk and psiTurk web services."""
+
+        # commit to the db so if anything goes wrong the rest of the
+        # request is safe
+        session.commit()
```

Comment:
tldr; the recruiter isnt commiting changes it has made, its commiting changes made by anything else prior to its instantiation.

The idea was that before the recruiter contacts AWS we need to commit whatever changes we have made to the db. Initially I started putting commit() throughout the recruiter code prior to any contact with AWS, but I realized that whenever the experiment makes contact with AWS it first builds a new recruiter object, therefore the cleanest way to make sure all prior changes to the db are committed is to have the recruiter commit on init. Given this, it would be a bad idea for an experiment to create a recruiter, then make a bunch of changes to the db, and then ask its recruiter to contact AWS. Better practice is to do whatever db operations you want and only then create and use the recruiter.